apps/discovery_api/test/unit/discovery_api_web/controllers/visualization_controller_test.exs

```diff
@@ -121,7 +114,8 @@ defmodule DiscoveryApiWeb.VisualizationControllerTest do
                "query" => @query,
                "title" => @title,
                "id" => @id,
-               "chart" => @decoded_chart
+               "chart" => @decoded_chart,
+               "allowed_actions" => [%{"name" => "update"}, %{"name" => "create_copy"}]
```

Comment:
Thank you for choosing this form for the return. We've chosen things like arrays of strings and maps before and it has locked us in pretty hard.

Reply:
👍 most of the credit to @JarredOlson